library/ctr_drbg.c

```diff
@@ -415,7 +409,8 @@ int mbedtls_ctr_drbg_seed_entropy_len(
     ctx->f_entropy = f_entropy;
     ctx->p_entropy = p_entropy;
 
-    ctx->entropy_len = entropy_len;
+    if( ctx->entropy_len == 0 )
```

Comment:
Hi @gilles-peskine-arm, is there any reason you didin't use `ctr_drbg_set_entropy_len()` here? Maybe you wanted to keep some level of method independency.

Reply:
I can't think of a strong reason one way or the other. Semantically, I want to set the `entropy_len` field to a certain value. `mbedtls_ctr_drbg_set_entropy_len()` does that, but so does an assignment. An assignment is simpler than calling a function. Arguably, calling a function is better in case the function changes. But changes to do what? If it changes to add some validation, this validation might not be what is needed at this point inside `mbedtls_ctr_drbg_seed`. And if the meaning of the field changes, there'll be other changes needed in the code, so who knows what would need to change in `mbedtls_ctr_drbg_seed`. In terms of code size, the assignment is a couple of bytes smaller than a function call. So I have a very slight preference for the direct assignment. But only very slight. I don't remember thinking about it explicitly at the time.

Reply:
Thanks for the perspective.